profilehooks.py

```diff
@@ -381,17 +409,24 @@ def print_stats(self):
 
     def reset_stats(self):
         """Reset accumulated profiler statistics."""
+        # send stats printing to specified stdout if it's file-like
+        stream = self.stdout if self._stdout_is_fp else sys.stdout
+
         # Note: not using self.Profile, since pstats.Stats() fails then
-        self.stats = pstats.Stats(Profile())
+        self.stats = pstats.Stats(Profile(), stream=stream)
         self.ncalls = 0
         self.skipped = 0
 
     def atexit(self):
-        """Stop profiling and print profile information to sys.stdout.
+        """Stop profiling and print profile information to sys.stdout or self.stdout.
 
         This function is registered as an atexit hook.
         """
-        self.print_stats()
+        if self._stdout_is_fp:
+            with redirect_stdout(self.stdout):
+                self.print_stats()
+        else:
+            self.print_stats()
```

Comment:
Curious: why do it this way instead of making self.print_stats() do `print(..., file=stream`) a few times?

AFAIU self.stats.print_stats() already prints to `stream`?

Reply:
I completely overlooked the `print(..., file=stream)` option, and you're right that `self.stats.print_stats()` will already go to the right place.

I'll make a quick port over to that method, it's simpler. I think I have a way to avoid repeating `file=stream` a bunch so as to make it hard for someone to change / add prints in `print_stats` and forget that kwarg.